### ISSUE TITLE
Export and import for templates/stories

### ISSUE THREAD

**Issue by marbd3** [google/web-stories-wp#4357]
## Feature Description
An export and import of the template feature is missing will it be added anytime soon?

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by ernee** [comment]
@marbd3 great suggestion! We've been discussing some export possibilities[ in this issue ](https://github.com/google/web-stories-wp/issues/3157) as well. 

How would you use an export/ import feature? 

Would it be similar to the request linked to export the assets used in the story or would it be more for importing/ exporting a story (template) from one WordPress site to another?

Feel free to elaborate and we'll share this with the team!

**Comment by theMikebal** [comment]
I think the challenge is that people are already building stories for other social channels from a mobile device or on another platform. Getting the assets from the cloud and then into WP is another step and then rebuilding the whole thing for web feels too duplicative. It would be great to see this functionality exist or work in a way where social media sharing tools or automation platforms like Zapier could allow users to take out some of the duplicity.

**Comment by dreamofabear** [comment]
This could also be generalized for stories to support one or multiple stories to be exported into a file and imported into a different WP instance.

- WP's standard export/import tool skips the `posts_content_filtered` field that contains story JSON data. The 3P plugins [WP Import Export Lite](https://wordpress.org/plugins/wp-import-export-lite/) and [WP All Export](https://wordpress.org/plugins/wp-all-export/) also appear to skip this field. Are there other WP export/import plugins or tools that could accomplish this?
- Exact usage requirements (e.g. user mapping, missing table references) are TBD.

**Comment by dreamofabear** [comment]
Relevant support thread: https://wordpress.org/support/topic/copy-story-to-another-instance/

**Comment by mxbclang** [comment]
Noting that we've received a few requests for this in support recently:

- https://wordpress.org/support/topic/export-and-import-20/
- https://wordpress.org/support/topic/export-import-of-web-stories-between-2-wp-website/

**Comment by mikezielonka** [comment]
Just adding in a unique edge.  I'm taking a 4 site Wordpress multisite to 1 single site.  It's been a real struggle finding a workflow to export and import the stories from the sites into the new instance.

**Comment by swissspidy** [comment]
@mikezielonka Thanks for sharing your experience. Curious, are you also struggling to finding a workflow for importing regular posts? If not, why is importing stories more difficult?

**Comment by mikezielonka** [comment]
> @mikezielonka Thanks for sharing your experience. Curious, are you also struggling to finding a workflow for importing regular posts? If not, why is importing stories more difficult?

WP All Export, the Core Exporter, and Migrate DB Pro don't offer a way to create an export that can be:

1 - Merged
2 - Include the `post_content_filtered` column with the [data needed from wp_posts and wp_postmeta ](https://wordpress.org/support/topic/export-and-import-20/#post-15078879)

**Comment by mikezielonka** [comment]
Did an Export using WP Import and WP All Export.  The great support from those guys helped me thru getting `posts_content_filtered` included in the migration.  I have the data there but the editor still shows blank.

Is this because the ID of the web story changed between staging and production?

**Comment by swissspidy** [comment]
Glad to hear you're making progress!

Did you also include `post_content` and post meta?

IDs shouldn't really matter to the plugin, but of course something else could have happened to trip up WP itself.

Hard to say without a DB dump or something.

**Comment by mikezielonka** [comment]
@swissspidy 

Here is a quick data dump.  I'd love to find a better way so that the many people who run into this have a basic workflow. 🤓 

Here is a DB dump of one row: https://s.mikezielonka.com/yAukkq9o

Sample editor url on prod `https://flavor-feed.com/wp-admin/post.php?post=3822&action=edit#page=%25221008f12a-d03d-4156-bad6-4a0a8c1b3900%2522`

Sample editor url on staging `https://flavor-feedcom.bigscoots-staging.com/wp-admin/post.php?post=3597&action=edit#page=%252295ab01c4-218f-402d-9b65-c5ec19b29a2b%2522`

Prod Web story https://flavor-feed.com/web-stories/sheet-pan-mac-and-cheese/

Staging Web Story https://flavor-feedcom.bigscoots-staging.com/web-stories/sheet-pan-mac-and-cheese/

*note* some of the images on prod likely are broken due to needing to regenerate thumbnails. :)

**Comment by mikezielonka** [comment]
@swissspidy any thoughts on what else I could attempt to get the web stories editor to work with the migrated stories?

**Comment by swissspidy** [comment]
It looks like the issue in your case is that the escaping of double quotes of the JSON content in the `post_content_filtered` column somehow got broken/missing.

You can test this yourself by taking the contents of the `post_content_filtered` column, unescaping it and running it through a JSON formatter/validator.

Because it's invalid JSON, it can't be parsed and thus the `story_data` column in the REST API will be empty (see https://flavor-feed.com/wp-json/web-stories/v1/web-story/3822?_fields=id,title,story_data for example)

Screenshots showing missing double quotes:

![Screenshot 2022-02-06 at 13 34 15](https://user-images.githubusercontent.com/841956/152680948-18c44cbb-c161-4598-aec9-a94d3f2ecf1f.png)
![Screenshot 2022-02-06 at 13 30 43](https://user-images.githubusercontent.com/841956/152680956-6d27e6f2-d78d-4bef-a4e8-f5862a0e879b.png)

Here's how it should be:

![Screenshot 2022-02-06 at 13 31 26](https://user-images.githubusercontent.com/841956/152680971-a3b31d15-34ea-491f-8b06-3fa68de19024.png)

Notice how the block quotes of the HTML attribute are escaped with a backslash.

**tl;dr:** Fix the JSON to make sure it's valid, then everything should be fine.

**Comment by mikezielonka** [comment]
> It looks like the issue in your case is that the escaping of double quotes of the JSON content in the `post_content_filtered` column somehow got broken/missing.
> 
> You can test this yourself by taking the contents of the `post_content_filtered` column, unescaping it and running it through a JSON formatter/validator.
> 
> Because it's invalid JSON, it can't be parsed and thus the `story_data` column in the REST API will be empty (see https://flavor-feed.com/wp-json/web-stories/v1/web-story/3822?_fields=id,title,story_data for example)
> 
> Screenshots showing missing double quotes:
> 
> ![Screenshot 2022-02-06 at 13 34 15](https://user-images.githubusercontent.com/841956/152680948-18c44cbb-c161-4598-aec9-a94d3f2ecf1f.png) ![Screenshot 2022-02-06 at 13 30 43](https://user-images.githubusercontent.com/841956/152680956-6d27e6f2-d78d-4bef-a4e8-f5862a0e879b.png)
> 
> Here's how it should be:
> 
> ![Screenshot 2022-02-06 at 13 31 26](https://user-images.githubusercontent.com/841956/152680971-a3b31d15-34ea-491f-8b06-3fa68de19024.png)
> 
> Notice how the block quotes of the HTML attribute are escaped with a backslash.
> 
> **tl;dr:** Fix the JSON to make sure it's valid, then everything should be fine.

Very Very odd.  I didn't even think to check the JSON.  Was able to fix that up and restore the story.  

If I'm able to pinpoint where that is coming from, I'll log an issue in the appropriate place. 

The other issue I ran into was with some images showing up as broken links.  That could be a side effect of the staging site.... still need to dig in there.

Thank you so much!

**Comment by mdotk** [comment]
> I think the challenge is that people are already building stories for other social channels from a mobile device or on another platform. Getting the assets from the cloud and then into WP is another step and then rebuilding the whole thing for web feels too duplicative. It would be great to see this functionality exist or work in a way where social media sharing tools or automation platforms like Zapier could allow users to take out some of the duplicity.

Following this issue along. The above sums it up nicely for me!

**Comment by mariyanh** [comment]
Has someone managed to export/import stories so far with some plugin?
Just to mention that I don't want to export/import the whole database...

Once I used some plugin for exporting custom post types and did the job but I don't remember it now.

Thanks...

**Comment by skimboardbr** [comment]
Hello
Any news?

I am very interested in exporting Stories from other tools/plugins and importing them into Google Web Stories

**Comment by swissspidy** [comment]
Importing from other story creation tools is not possible as all these tools use different data formats.

**Comment by miguelangel812** [comment]
It would be great to have this product upgrade. Thank you!

**Comment by JorgeVilchez95** [comment]
Hi! I would love to see this functionality added, I need it. Thank you!

**Comment by swissspidy** [comment]
If you‘re interested in seeing this feature, please add an emoji reaction to the original issue description at the top instead of leaving comments. It‘s less noisy.

**Comment by sourcer18** [comment]
> If you‘re interested in seeing this feature, please add an emoji reaction to the original issue description at the top instead of leaving comments. It‘s less noisy.

nice job Pascal!

**Comment by bmattb** [comment]
Added the NC label as this is something that team is also now requesting

**Comment by wmosquini2** [comment]
I was able to export WebStories with a little help from Adam.

I had already managed to import the content, but the Stories were not displayed in the editor, even though the content was there. That's where Adam gave the light at the end of the tunnel commenting that the Wordpress export tool doesn't export the "post_content_filtered" bank column. To solve it, I created a query to export the column data from the old bank and imported it into the new one. Of course, it wasn't 100%, but it worked on 98% of the Stories created.

It would be very helpful to include this native export/import tool in the plugin. This is extremely important in cases where it is necessary to create a new installation of Wordpress. Despite being a simple solution, it required good hours of work to reach the final result.

**Comment by eo48Oakes** [comment]
> I was able to export WebStories with a little help from Adam.
> 
> I had already managed to import the content, but the Stories were not displayed in the editor, even though the content was there. That's where Adam gave the light at the end of the tunnel commenting that the Wordpress export tool doesn't export the "post_content_filtered" bank column. To solve it, I created a query to export the column data from the old bank and imported it into the new one. Of course, it wasn't 100%, but it worked on 98% of the Stories created.
> 
> It would be very helpful to include this native export/import tool in the plugin. This is extremely important in cases where it is necessary to create a new installation of Wordpress. Despite being a simple solution, it required good hours of work to reach the final result.

Hello, can you please share that query you used? I would be so much happy. Thanks in Advance!!

**Comment by eo48Oakes** [comment]
> I was able to export WebStories with a little help from Adam.
> 
> I had already managed to import the content, but the Stories were not displayed in the editor, even though the content was there. That's where Adam gave the light at the end of the tunnel commenting that the Wordpress export tool doesn't export the "post_content_filtered" bank column. To solve it, I created a query to export the column data from the old bank and imported it into the new one. Of course, it wasn't 100%, but it worked on 98% of the Stories created.
> 
> It would be very helpful to include this native export/import tool in the plugin. This is extremely important in cases where it is necessary to create a new installation of Wordpress. Despite being a simple solution, it required good hours of work to reach the final result.

Hello @swissspidy,  please help me to get that query or any alternative method to export/import web stories so that the webstories could be editable. 

Thanks

**Comment by mikezielonka** [comment]
@swissspidy has anything changed with how web stories are storing data in the WP db?

**Comment by mikezielonka** [comment]
@swissspidy  has anything changed with how web stories are storing data in the WP db?

I migrated the `post_content_filtered` content from a staging site to prod like I've done in the past.  I can see the data is safely in the DB for the correct post ID but the editor loads an empty story.

BUT it does display correctly when viewing the story.

**Comment by swissspidy** [comment]
No, nothing has changed. Your JSON data in the db might be corrupt or something. Please open a new issue if you need assistance with that.

**Comment by mikezielonka** [comment]
> No, nothing has changed. Your JSON data in the db might be corrupt or something. Please open a new issue if you need assistance with that.

Better to open an issue here or on WP dot org?

I confirmed my json was valid several times :)

**Comment by swissspidy** [comment]
Here on GitHub is fine. Just so we don't add unnecessary noise to this feature request issue here.

So please open a new GH issue with as much information as possible, ideally including a db dump for that specific story, or at least the story URL, so that we have something to work with.

**Comment by kaushik0x7d2** [comment]
Hey @swissspidy, is there any update on this feature?

**Comment by kaushik0x7d2** [comment]
Import custom templates, able to edit it and export the code to add to the website.
